### PR TITLE
Support planet "language" and attributes "no trading", "no jobs", "no banking", and "no hiring".

### PR DESCRIPTION
**Pull request by jmhorjus**
A planet can now have a "language" (i.e. "language Esperanto")  which will automatically disable banking, hiring, and jobs unless the player condition "language: Esperanto" is defined.  If a planet does not define a language, the language of the planet's government will be used (or the language of the system's government, if the planet does not define a government).  Planets need not define languages individually; normally language will still be defined only by the government, and you will still get the full effect.  Setting "language clear" on a planet can disable any language requirement for banking/hiring/jobs that might have been inherited from a planet/system government. 

Trading, outfitter, and shipyard availability is unaffected by language (universal currency or barter economy I guess?).  

Planets can also set "no trading", "no jobs", "no banking", or "no hiring" as attributes.  By default, all these things are still all available on inhabited planets that have a spaceport, just as they were before.

### CODE REVIEW

File: source/Planet.cpp [hunk 1]
```diff
@@ -218,7 +218,39 @@ bool Planet::IsInhabited() const
 }
 
 
-	
+
+// Check if this planet has a bank (i.e. is inhabited and does not have "no trading" attribute)
+bool Planet::HasTrading() const
```
Comment (luiges90) [inline source/Planet.cpp]: Looks like copy-pasting error in the comments :sweat_smile:

File: source/Planet.cpp [hunk 2]
```diff
@@ -218,7 +218,39 @@ bool Planet::IsInhabited() const
 }
 
 
-	
+
+// Check if this planet has a bank (i.e. is inhabited and does not have "no trading" attribute)
```
Comment (luiges90) [inline source/Planet.cpp]: Looks like copy-pasting error in the comments :sweat_smile:

**Comment by jmhorjus** [comment]
@luiges90 Thanks.  Comment fixed.

**Comment by LocalGod79** [comment]
i guess this is where i come in and add these attributes to each planet in the map file?

**Comment by LocalGod79** [comment]
#666 ?

**Comment by jmhorjus** [comment]
After adding this, I was thinking of the Wanderers and how the mission already sets a "language: Wanderers" condition which is then never used.  

What if you could also set "language ABC" on a planet, which would be shorthand for "No bank, trading, jobs, or hiring here unless you have a matching "language: ABC" condition set in your player."
That would avoid the necessity to set all the attributes on the planets individually and then have events to remove them.  And if any other alien stories involved language in the future it would same some trouble in that process as well.  

Of course being able to set "no banking", "no hiring", etc individually on planets for whatever reason would still be valuable.  But for the Wanderers, having the game engine natively support "language" would make the data files simpler.

**Comment by LocalGod79** [comment]
agreed.

i believe the "language: Wanderers" condition is an indication of M.Z.'s intention to use it with a variety of species.

your proposed solution (the addendum to the also-desired function added the the pull) sounds elegant and promising.

**Comment by jmhorjus** [comment]
"Language" support added and tested with Wanderers.

**Comment by jmhorjus** [comment]
I think perhaps based on the new Korath content, that trading should not require you to know the planet's language by default - only jobs, hiring, and banking should be disabled if you don't know a planet's language. The assumption seems to be that at first you are able to trade with some factions of Korath, even without being able to understand their speech.

**Comment by wambat1** [comment]
so, if a pilot can speak Wanderer, and lands on a Wanderer planet and tops up his crew, does this mean the player now has wanderer crew members?

**Comment by jmhorjus** [comment]
Yep, the assumption is that if you can speak wanderer you can hire wanderer crew. :)

Makes much more sense than being able to hire them without speaking their language. You can disable hiring specifically using "no hiring" if you want a planet not to allow hiring regardless of language.

**Comment by wambat1** [comment]
on some other alien worlds i could make the assumption that i was hiring humans on the planet incidentally. i think the Hai would also be plausible crew members, though that assumption conflicts with the Wanderer diplomat mission. in the Kor and Wanderer regions this assumption seems quite improbable. it seems an even greater problem for pilots who visit the Pug worlds before they start annexing human systems; you could be flying around with a bunch of EMHs for crew. 

    On Monday, May 23, 2016 8:45 AM, jmhorjus <notifications@github.com> wrote:
 

 Yep, the assumption is that if you can speak wanderer you can hire wanderer crew. :)Makes much more sense than being able to hire them without speaking their language. You can disable hiring specifically using "no hiring" if you want a planet not to allow hiring regardless of language. —

**Comment by jmhorjus** [comment]
@wambat1 Yeah, I'd agree wanderer space should probably not have hiring right away even after you get translation. 

However, that is a content decision for another pull request.  This change is intended to simply add support for the _ability_ to disable hiring and/or other services individually on planets without removing the spaceport or making the place "uninhabited".  

By default, you can never hire on planets where you can't communicate, but once you know the language it would have to be a specific content decision to add "no hiring" to a planet.

**Comment by wambat1** [comment]
shall i start a new pull for content?

**Comment by jmhorjus** [comment]
Merged with the latest master; had to resolve some conflicts with this commit:
 https://github.com/endless-sky/endless-sky/commit/a10992fe0815568a2efe3c2fa32ccb53263cfef7

Pretty sure I got it right, but I don't know how to reproduce the crash that that other commit was fixing, so I haven't re-validated that specifically.

**Comment by jmhorjus** [comment]
@wambat1 Yeah, putting something together for content would be great.  You can branch from my branch now if you want since it's up to date with endless-sky/master - and then when this is eventually merged your pull could be right behind it without conflicts.

**Comment by jmhorjus** [comment]
Hmm...governments also define languages...do you think that planets that don't define a language should automatically inherit the language of their government?

**Comment by jmhorjus** [comment]
Tested this in korath space and in wanderer space, going through the whole "language" quest chain and observing that everything works well.  

Defaulting to the language of the planet's government in a big plus here, since in the current data files only governments define languages - so this avoids the need to change any existing data.

**Comment by jmhorjus** [comment]
@endless-sky Any comment? I think this would be very nice for modders and should be quite safe. Only thing is it should probably be accompanied by a small documentation update (something similar to the one-paragraph description of this pull request).

**Comment by jmhorjus** [comment]
Um...anybody there?

**Comment by jmhorjus** [comment]
Closing: Superseded by #2543.

**Pull request closed without merging**